rfcs/framework-0063-notification-system.md

```diff
@@ -0,0 +1,422 @@
+- Start Date: 2022-06-13
+- RFC PR: [#63](https://github.com/inveniosoftware/rfcs/pull/63)
+- Authors: David Eckhard, Zacharias Zacharodimos, Alex Ioannidis
+
+# Notification System
+
+## Summary
+
+Implementation details for a notification system to be used in InvenioRDM.
+
+## Motivation
+
+As the product is getting larger and new features are added, it is hard to keep an overview of everything going on as a user. With the communities feature, certain tasks require an action to be executed by the user (f.e. accepting an incoming request for a community). This incoming request can easily get "lost", if a user does not explicitly check the community dashboard. Therefore, a notification system will not only solve this problem (and similar ones) but also increase user engagement by advertising the repository some more.
+
+It will take care of sending the notifications based on the configuration and user preferences (i.e. if the repository disables certain notifications or a user does not want to receive certain notifications, they should not be sent).
+
+### Use Case
+
+- Send notifications for operations involving multiple users (see [table](#notification-examples))
+- Overridable notification templates
+  - One template for each notification
+  - Can be overridden via theme (like other templates)
+  - Add new templates for custom notifications
+  - Should be translatable
+    - Store preferred language in user profile
+- Modify notifications
+  - Set which notifications should be send
+    - As a repository admin activate/deactive certain notifications
+    - As a user (user profile)
+  - Define notifications for events per module
+  - Define notification backends per module
+
+## Detailed design
```

Comment:
I'm missing statements about which layer which parts belong to in the software architecture: https://inveniordm.docs.cern.ch/develop/architecture/software/